app/geant-exporter/geant-exporter.cc

```diff
@@ -0,0 +1,246 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geant-exporter.cc
+//---------------------------------------------------------------------------//
+/*! \brief Geant4 particle definition and physics tables exporter
+ *
+ *  The code exports particle information and physics tables built by the
+ *  selected physics list and geometry. Output data is stored into ROOT files
+ *  and read by Celeritas' Geant Importer class.
+ */
+//---------------------------------------------------------------------------//
+
+// C++
+#include <string>
+#include <vector>
+
+// Project
+#include "src/DetectorConstruction.hh"
+#include "src/ActionInitialization.hh"
+#include "src/GeantPhysicsTableParser.hh"
+
+// Geant4
+#include "G4RunManager.hh"
+#include "G4UImanager.hh"
+//#include "FTFP_BERT.hh"
+#include "G4VModularPhysicsList.hh"
+#include "G4GenericPhysicsList.hh"
+#include "G4PhysicsTable.hh"
+#include "G4VEmProcess.hh"
+#include "G4VEnergyLossProcess.hh"
+
+// ROOT
+#include "TFile.h"
+#include "TDirectory.h"
+#include "TTree.h"
+#include "TBranch.h"
+
+int main(int argc, char** argv)
+{
+    //-----------------------------------------------------------------------//
+    /*!
+     * Minimal help
+     */
+    if (argc == 1 || argc != 2)
+    {
+        G4cout << "Usage: ./geant4-exporter geometry.gdml" << G4endl;
+        return 0;
+    }
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Starting the run manager
+     */
+    auto run_manager = new G4RunManager;
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Fetching the GDML input file and initializing the geometry
+     */
+    std::string gdml_input = argv[argc - 1];
+
+    run_manager->SetUserInitialization(new DetectorConstruction(gdml_input));
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Loading the physics list
+     */
+    std::vector<G4String>* physics_constructor = new std::vector<G4String>;
+    physics_constructor->push_back("G4EmStandardPhysics");
+
+    G4VModularPhysicsList* physics_list;
+
+    // Adding the user-defined physics list
+    physics_list = new G4GenericPhysicsList(physics_constructor);
```

Comment:
I will test this. 

I am not sure how Geant behaves with a unique_ptr (i.e. I don't know if the pointers to the virtual classes are copied internally). Documentation suggests to use raw pointers and that the run manager is responsible for deleting everything, so that is where the decision comes from.

Reply:
The code I posted above should generate a completely identical program to what you have; the difference is that it's "safer" (cleaning up automatically if you forget to call `release`) and easier to read (it's 100% explicit about who we expect will take care of the memory cleanup). For someone like me who doesn't know Geant super well, I wouldn't necessarily be able to tell that geant is taking ownership of the pointer.